src/d/actor/d_a_obj_hat.cpp

```diff
@@ -4,91 +4,241 @@
 //
 
 #include "d/actor/d_a_obj_hat.h"
+#include "d/actor/d_a_npc_roten.h"
+#include "d/d_a_obj.h"
+#include "d/d_com_inf_game.h"
+#include "d/d_kankyo.h"
 #include "d/d_procname.h"
+#include "d/res/res_ro.h"
+#include "JSystem/J3DGraphAnimator/J3DModel.h"
+
+static const int l_bmd_ix_tbl[] = {RO_BDL_RO_HAT, RO_BDL_RO_HAT2, RO_BDL_RO_HAT3, RO_BDL_RO_HAT};
+static const int l_bck_ix_tbl[] = {RO_INDEX_BCK_HAT_WID, RO_BCK_HAT2_WIND, RO_BCK_HAT3_WID,
```

Comment:
The BCK list shouldn't be using the `INDEX` enum here, as its being passed to `dComIfG_getObjectIDRes` later. `RO_RES_FILE_ID` gets passed to `dComIfG_getObjectIDRes`, while `RO_RES_FILE_INDEX` gets passed to dComIfG_getObjectRes`.